cpp/api-docs.rst

```diff
@@ -499,10 +461,73 @@ For internal consistency, always use ``@see`` and not the synonymous ``@sa``.
 Notes
 -----
 
-*Notes* is an optional section that provides additional information about the code, possibly including a discussion of the algorithm or known limitations of the code. The notes must be prefixed by a ``@note`` or ``@warning`` command. Equations or images may be used as described in :ref:`cpp-doxygen-extended-summary`.
+*Notes* is an optional section that provides additional conceptual information about the API.
 
+The notes must be prefixed by a ``@note`` or ``@warning`` command.
 For internal consistency, always use ``@note`` and not the synonymous ``@remark`` or ``@remarks``.
 
+Some things to include in a *Notes* section:
+
+- Discussions of features, going beyond the level of the :ref:`summary sentence <py-docstring-short-summary>` and :ref:`extended summary <py-docstring-extended-summary>`.
+- Usage patterns, like how code is expected to use this API, or how this API is intended to be used in relation to other APIs.
+- Background theory. For example, if the API implements an algorithm, you can fully document the algorithm here.
+- Implementation details and limitations, if those details affect the user's experience.
+  Purely internal details should be written as regular code comments.
+
+Specific how-tos, tutorials, and examples go in the :ref:`Examples section <py-docstring-examples>` instead of *Notes*.
+The *Notes* section is dedicated to conceptual documentation.
+
+The :ref:`cpp-doxygen-parameters` and :ref:`cpp-doxygen-returns` sections are the best places to describe specific input and output variables.
+The *Notes* section can still reference these variables by name (see :ref:`py-docstring-parameter-markup`), and discuss how they work at a big-picture level.
+Since the content in the :ref:`cpp-doxygen-parameters` needs to be brief, you can write additional content as part of the *Notes* section.
```

Comment:
BTW, there's still a note in the Parameters section that says to use Extended Discussion, not Notes. Please update it to match.